 07-verbose-apigen.diff

 Make the apigen stage of the build actually say something; could timeout
 on slow buildds/is boring, etc. Patch by Chris Lamb <chris@chris-lamb.co.uk>.

diff -urNad py-0.9.1.orig/py/bin/_docgen.py py-0.9.1/py/bin/_docgen.py
--- py-0.9.1.orig/py/bin/_docgen.py	2008-07-04 15:35:50.000000000 +0100
+++ py-0.9.1/py/bin/_docgen.py	2008-07-04 15:35:54.000000000 +0100
@@ -17,7 +17,7 @@
     cmd = ('PYTHONPATH="%s" %s python "%s" %s "%s"' %
             (pypath.dirpath(), envvars, pytestpath, args, path))
     print cmd
-    py.process.cmdexec(cmd)
+    py.process.cmdexec(cmd, debug=True)
 
 def build_apigen_docs(targetpath, testargs=''):
     run_tests(pypath,
diff -urNad py-0.9.1.orig/py/process/cmdexec.py py-0.9.1/py/process/cmdexec.py
--- py-0.9.1.orig/py/process/cmdexec.py	2008-07-04 15:35:50.000000000 +0100
+++ py-0.9.1/py/process/cmdexec.py	2008-07-04 15:36:24.000000000 +0100
@@ -16,7 +16,7 @@
 #-----------------------------------------------------------
 # posix external command execution
 #-----------------------------------------------------------
-def posix_exec_cmd(cmd):
+def posix_exec_cmd(cmd, debug=False):
     """ return output of executing 'cmd'.
 
     raise ExecutionFailed exeception if the command failed.
@@ -78,6 +78,8 @@
             if not data:
                 r.close()
                 continue
+            if debug:
+                sys.stderr.write(data)
             if r is stdout:
                 out.append(data)
             else:
